index.js

```diff
@@ -448,6 +449,43 @@ module.exports = function(options) {
     return self.callAllAndEmit('afterInit', 'afterInit', callback);
   }
 
+  function lintModules(callback) {
+    _.each(self.modules, function(module, name) {
+      if (name.match(/-widgets$/) && (!extending(module)) && (!module.options.fromScratch)) {
+        lint('The module ' + name + ' does not extend anything.\n\nA widgets module usually extends `apostrophe-widgets` or\n`apostrophe-pieces-widgets`. Or possibly you forgot to npm install something.\n\nIf you are sure you are doing the right thing, set the `fromScratch` option\nto `true` for this module.');
+      } else if (name.match(/-pages$/) && (name !== 'apostrophe-pages') && (!extending(module)) && (!module.options.fromScratch)) {
+        lint('The module ' + name + ' does not extend anything.\n\nA -pages module usually extends `apostrophe-custom-pages` or\n`apostrophe-pieces-pages`. Or possibly you forgot to npm install something.\n\nIf you are sure you are doing the right thing, set the `fromScratch` option\nto `true` for this module.');
+      } else if ((!extending(module)) && (!hasConstruct(name)) && (module.options.hasCode !== false)) {
```

Comment:
`apostrophe-seo` and `apostrophe-open-graph` threw this warning, seemingly because they're bundles. Maybe `moogBundle` should prevent this warning as well.

Reply:
Working on this, good catch.